bin/ruby-build

```diff
@@ -342,6 +342,10 @@ fetch_tarball() {
   local package_filename="${package_name}.tar.gz"
 
   if [ "$package_url" != "${package_url%bz2}" ]; then
+    if [ -z "$(which bzip2)" ]; then
```

Comment:
This check could be `if ! type -p bzip2 >/dev/null; then`. It's the way we do such checks elsewhere in the codebase.